app/src/main/java/net/bible/android/view/util/buttongrid/ButtonGrid.kt

```diff
@@ -218,14 +212,13 @@ class ButtonGrid constructor(context: Context, attrs: AttributeSet? = null, defS
 
     private fun showPreview(buttonInfo: ButtonInfo) {
         try {
-            val button = buttonInfo.button
             if (buttonInfo != currentPreview) {
-                Log.d(TAG, "Previewing " + buttonInfo.name)
+                Log.d(TAG, "Previewing " + buttonInfo.description)
                 currentPreview = buttonInfo
-                previewText!!.text = buttonInfo.name
+                previewText!!.text = buttonInfo.description
```

Comment:
The double exclamation `!!` should not be used normally. It causes a crash when there's null, and especially in cases like this where it's not some important app logic, it should never cause a crash (even though it should never be null). I see you just used what was there from before and that's not a problem 😄 

It would be better to add a new line under `Log.d` (line 216), like this:
`val preview = previewText ?: return`

And then now use `preview.` everywhere underneath where there previously was `previewText!!.`

Reply:
Yes my bad. I use to follow boy scout rule but I forgot for this class. Looks like it was a Java class automatically converted to Kotlin and there are plenty of Java-ish remains to be reviewes :sweat_smile: